src/sync/sync_session.cpp

```diff
@@ -283,12 +299,147 @@ SyncSession::SyncSession(SyncClient& client, std::string realm_path, SyncConfig
 , m_realm_path(std::move(realm_path))
 , m_client(client) { }
 
+std::string SyncSession::get_recovery_file_path()
+{
+    std::time_t now = std::time(nullptr);
+    std::stringstream stream;
+    stream << SyncManager::shared().recovery_directory_path();
+    stream << "recovered_realm-" << std::put_time(std::localtime(&now), "%Y%m%d-%H%M%S") << "-XXXXXXXX";
+    std::string path = stream.str();
+    // Now try to create the temporary file.
+    int fd = mkstemp(&path[0]);
```

Comment:
Core has `make_temp_dir()` which might be useful. `mkstemp` might be better than core's `mkdtemp`.